nix/cabal-project.nix

```diff
@@ -0,0 +1,159 @@
+# editorconfig-checker-disable-file 
+
+{ repoRoot, inputs, lib, ... }:
+
+let
+  cabalProjectArgs = { config, pkgs, ... }: {
+
+    name = "plutus";
+
+    compiler-nix-name = lib.mkDefault "ghc927";
+
+    src = ../.;
+
+    # We would expect R to be pulled in automatically as it's a dependency of
+    # plutus-core, but it appears it is not, so we need to be explicit about
+    # the dependency on R here. Adding it as a buildInput will ensure it's
+    # added to the pkg-config env var.
+    shell = {
+      withHoogle = false;
+      buildInputs = [ pkgs.R ];
+    };
+
+    flake.variants = {
```

Comment:
Personally I don't like the flake variants. I don't understand why people want to do `f(args).adjustFToHaveDifferentArgs` rather than `f(differentArgs)`. It's really not hard, and it's super hard to work out what's going on with all the clever variants and overriding stuff. What's Wrong With Functions, Damnit. But I've made enough of a fuss, I won't fight it.